crates/fee/src/lib.rs

```diff
@@ -444,24 +418,9 @@ impl<T: Config> Pallet<T> {
         Ok(())
     }
 
-    fn distribute<
-        Currency: TryInto<SignedInner>,
-        SignedInner: TryInto<Currency> + CheckedDiv,
-        SignedFixedPoint: FixedPointNumber<Inner = SignedInner>,
-        Rewards: reward::Rewards<T::AccountId, SignedFixedPoint = SignedFixedPoint>,
-    >(
-        reward: Currency,
-    ) -> Result<Currency, DispatchError> {
-        let reward_as_inner = reward.try_into().ok().ok_or(Error::<T>::TryIntoIntError)?;
-        let reward_as_fixed =
-            SignedFixedPoint::checked_from_integer(reward_as_inner).ok_or(Error::<T>::TryIntoIntError)?;
-        Ok(Rewards::distribute_reward(reward_as_fixed)?
-            .into_inner()
-            .checked_div(&SignedFixedPoint::accuracy())
-            .ok_or(Error::<T>::ArithmeticUnderflow)?
-            .try_into()
-            .ok()
-            .ok_or(Error::<T>::TryIntoIntError)?)
+    fn distribute(reward: &Amount<T>) -> Result<Amount<T>, DispatchError> {
+        let leftover = T::VaultRewards::distribute_reward(reward.to_fixed()?)?;
```

Comment:
Should we verify here that the `CurrencyId` of the `Amount` is the rewards currency?